tests/test_utils.py

```diff
@@ -1026,6 +1028,50 @@ def test_get_label_from_display_name(self, test_dn: str, data_model_labels: str)
                 return
         return
 
+
+class TestExportGraph:
+    def test_export_graph_success(self, tmp_path):
+        # Create a temporary file path
+        file_path = tmp_path / "graph.pickle"
+
+        # Define a sample schema
+        schema = {
+            "node1": {"edges": ["node2", "node3"]},
+            "node2": {"edges": []},
+            "node3": {"edges": []},
+        }
+
+        # Call the export_graph function
+        export_graph(schema, str(file_path))
+
+        # Check if the file exists
+        assert file_path.exists()
+
+        # Load the saved schema from the file
+        with open(file_path, "rb") as file:
+            saved_schema = pickle.load(file)
+
+        # Check if the saved schema is equal to the original schema
+        assert saved_schema == schema
+
+    def test_export_graph_failure(self, tmp_path, caplog):
+        # Create a temporary file path
+        file_path = tmp_path / "graph.pickle"
+
+        # Define a sample schema
+        schema = {
+            "node1": {"edges": ["node2", "node3"]},
+            "node2": {"edges": []},
+            "node3": {"edges": []},
+        }
+
+        # Set the file path to a non-existent directory
+        invalid_file_path = str(tmp_path / "non_existent_directory" / "graph.pickle")
+
+        # Call the export_graph function with an invalid file path
+        with pytest.raises(Exception):
```

Comment:
Are you able to capture a more specific Exception that is raised here? Is this like an IOException?